README.md

```diff
@@ -100,6 +100,89 @@ On the right is the Central E-SOH API Endpoint. In the middle are all the compon
+
+The relational database approach is similar to the document database approach (See above), the only change will be the actual database itself. 
+​
+![](https://raw.githubusercontent.com/EURODEO/e-soh-c4/main/04-component-diagrams/datastore-options/c4-container-relational-data-store-conventional.png)
+​
+##### Relational database with JSON capability
+​
+E.g. PostgreSQL and MySQL support JSON datatype. That makes it possible to support structured and semi-structured data in relational database by adding a column for JSON data. 
+​
+The relational database with JSON capability will bridge the fast column lookups from the relational database with the flexibility from the document database approach. This hybrid model is built by the same components as in the relational database (See previous subsection), the only difference is the special JSON datatype column. Probably JSON-queries in pure document-based database will be faster than this hybrid model, but we still maintain the speed from the relational columns.
+
+##### Files-based storage
+​
+Files-based storage was considered, but as query logic has to be implemented separately and there is requirements for structured data and queries, we do not consider it as a feasible option for implementation. There's also another downside with files which is challenge to update minor changes (e.g. incoming QC updates some minutes after the initial observation time) in the middle of the files. When using files, there are typically also some challenges with locking while writing -- in databases there are transactions to handle this// 2c
+ better for multiple calls.
+​
+If files-based datasets, e.g. 2d/3d imaging data or similar, needs to be handled, the support can be added e.g. by integrating S3 object storage into the system and using references to file URLs from the database. Database system will then serve also as a (discovery) metadata backend for files.
+
+
+#### Event sourcing architecture scenario
+
+The specification included a requirement for the Replay functionality, which means that a  user can request missed notification messages for a certain period. Thus we looked into an architecture which needed to have all notifications stored in some way
+
+If we assume that the notification messages also have the actual data (and metadata),
+an [event sourcing](https://www.martinfowler.com/eaaDev/EventSourcing.html) architecture might be appropriate:
+
+- All incoming data is processed and translated into the notification event format, probably some kind of data.
+- All these events are stored in an event store. This is the source of truth.
+- The Replay API is built on top of the event store, and simply returns all notifications in a specific time range.
+- The events are also ingested into the "view" database, which is designed to allow efficient EDR queries.
```

Comment:
What is the "view" database?

Reply:
It's referencing the PostGIS DB in the model where there are both Elastic + PostGIS. But now we know that also Elastic can offer geo-queries, so two DB model might become into one DB model also in Elastic case.